### PR TITLE
Send basic inventory with RegisterAgent

### PR DESCRIPTION
**Pull request by adjackura**
Make sure we run RegisterAgent immediately on agent start and agent enablement
Seperate internal agent periodics (just restart checker for now) from the service specifc ones
Update some core library dependencies

### CODE REVIEW

File: agentendpoint/agentendpoint.go [hunk 1]
```diff
@@ -101,7 +102,20 @@ func (c *Client) RegisterAgent(ctx context.Context) error {
 		return err
 	}
 
-	req := &agentendpointpb.RegisterAgentRequest{AgentVersion: agentconfig.Version(), SupportedCapabilities: agentconfig.Capabilities()}
+	oi, err := osinfo.Get()
```
Comment (jaiminsh) [inline agentendpoint/agentendpoint.go:105]: Should we populate this only if inventory is turned on?

File: main.go [hunk 2]
```diff
@@ -170,6 +170,10 @@ func runTaskLoop(ctx context.Context, c chan struct{}) {
 		// Set debug logging settings so that customers don't need to restart the agent.
 		logger.SetDebugLogging(agentconfig.Debug())
 		clog.DebugEnabled = agentconfig.Debug()
+		if agentconfig.TaskNotificationEnabled() && taskNotificationClient == nil {
+			// Call RegisterAgent now since we just we either just started running or were just enabled.
+			registerAgent(ctx)
```
Comment (jaiminsh) [inline main.go:175]: Not blocking this PR, but we should also call RegisterInventory here. Something to track

Comment (adjackura) [reply main.go:175]: Inventory happens on it's own schedule (default once every 10 min) so between agent being enabled and an inventory run will be <10min which is totally fine.

File: agentendpoint/agentendpoint.go (hunk 1 above)
Comment (adjackura) [reply agentendpoint/agentendpoint.go]: Fixed

**Pull request merged**